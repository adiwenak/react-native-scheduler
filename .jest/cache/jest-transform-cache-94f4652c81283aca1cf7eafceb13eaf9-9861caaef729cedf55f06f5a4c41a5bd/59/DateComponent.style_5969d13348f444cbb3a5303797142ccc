1bfe167678ee86e224ead1f13cb04af7
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _reactNative = require("react-native");

var _default = _reactNative.StyleSheet.create({
  topOuterView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  selectedColor: {
    backgroundColor: "#FF1493"
  },
  circleView: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  textStyleWeekday: {
    color: "#323A48",
    fontWeight: "500",
    fontSize: 14
  },
  textStyleWeekend: {
    color: "#768398",
    fontWeight: "400",
    fontSize: 14
  }
});

exports.default = _default;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi9Vc2Vycy9ha3VybmFkaS9Qcm9qZWN0cy9yZWFjdC1uYXRpdmUtc2NoZWR1bGVyL2xpYnJhcnkvQ2FsZW5kYXJNb250aC9EYXRlQ2FsZW5kYXJCb3gvSW5kaXZpZHVhbENvbXBvbmVudHMvRGF0ZUNvbXBvbmVudC5zdHlsZS50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7OztBQUFBOztlQVVlLHdCQUFXLE1BQVgsQ0FBa0I7QUFDN0IsRUFBQSxZQUFZLEVBQUU7QUFDVixJQUFBLElBQUksRUFBRSxDQURJO0FBRVYsSUFBQSxjQUFjLEVBQUUsUUFGTjtBQUdWLElBQUEsVUFBVSxFQUFFO0FBSEYsR0FEZTtBQU03QixFQUFBLGFBQWEsRUFBRTtBQUNYLElBQUEsZUFBZSxFQUFFO0FBRE4sR0FOYztBQVM3QixFQUFBLFVBQVUsRUFBRTtBQUNSLElBQUEsSUFBSSxFQUFFLENBREU7QUFFUixJQUFBLGNBQWMsRUFBRSxRQUZSO0FBR1IsSUFBQSxVQUFVLEVBQUU7QUFISixHQVRpQjtBQWM3QixFQUFBLGdCQUFnQixFQUFFO0FBQ2QsSUFBQSxLQUFLLEVBQUUsU0FETztBQUVkLElBQUEsVUFBVSxFQUFFLEtBRkU7QUFHZCxJQUFBLFFBQVEsRUFBRTtBQUhJLEdBZFc7QUFtQjdCLEVBQUEsZ0JBQWdCLEVBQUU7QUFDZCxJQUFBLEtBQUssRUFBRSxTQURPO0FBRWQsSUFBQSxVQUFVLEVBQUUsS0FGRTtBQUdkLElBQUEsUUFBUSxFQUFFO0FBSEk7QUFuQlcsQ0FBbEIsQyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IFN0eWxlU2hlZXQsIFRleHRTdHlsZSwgVmlld1N0eWxlIH0gZnJvbSBcInJlYWN0LW5hdGl2ZVwiXG5cbmludGVyZmFjZSBTdHlsZSB7XG4gICAgdG9wT3V0ZXJWaWV3OiBWaWV3U3R5bGVcbiAgICBzZWxlY3RlZENvbG9yOiBWaWV3U3R5bGVcbiAgICBjaXJjbGVWaWV3OiBWaWV3U3R5bGVcbiAgICB0ZXh0U3R5bGVXZWVrZGF5OiBUZXh0U3R5bGVcbiAgICB0ZXh0U3R5bGVXZWVrZW5kOiBUZXh0U3R5bGVcbn1cblxuZXhwb3J0IGRlZmF1bHQgU3R5bGVTaGVldC5jcmVhdGUoe1xuICAgIHRvcE91dGVyVmlldzoge1xuICAgICAgICBmbGV4OiAxLFxuICAgICAgICBqdXN0aWZ5Q29udGVudDogXCJjZW50ZXJcIixcbiAgICAgICAgYWxpZ25JdGVtczogXCJjZW50ZXJcIixcbiAgICB9LFxuICAgIHNlbGVjdGVkQ29sb3I6IHtcbiAgICAgICAgYmFja2dyb3VuZENvbG9yOiBcIiNGRjE0OTNcIlxuICAgIH0sXG4gICAgY2lyY2xlVmlldzoge1xuICAgICAgICBmbGV4OiAxLFxuICAgICAgICBqdXN0aWZ5Q29udGVudDogXCJjZW50ZXJcIixcbiAgICAgICAgYWxpZ25JdGVtczogXCJjZW50ZXJcIixcbiAgICB9LFxuICAgIHRleHRTdHlsZVdlZWtkYXk6IHtcbiAgICAgICAgY29sb3I6IFwiIzMyM0E0OFwiLFxuICAgICAgICBmb250V2VpZ2h0OiBcIjUwMFwiLFxuICAgICAgICBmb250U2l6ZTogMTRcbiAgICB9LFxuICAgIHRleHRTdHlsZVdlZWtlbmQ6IHtcbiAgICAgICAgY29sb3I6IFwiIzc2ODM5OFwiLFxuICAgICAgICBmb250V2VpZ2h0OiBcIjQwMFwiLFxuICAgICAgICBmb250U2l6ZTogMTRcbiAgICB9XG59KVxuIl19