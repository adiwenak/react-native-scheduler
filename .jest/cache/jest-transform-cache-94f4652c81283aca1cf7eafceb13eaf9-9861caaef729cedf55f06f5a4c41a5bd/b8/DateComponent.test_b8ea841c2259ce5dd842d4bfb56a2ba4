2de432e462a627dbab03b3736f089832
var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var React = _interopRequireWildcard(require("react"));

var _reactTestRenderer = require("react-test-renderer");

var _DateComponent = require("./DateComponent");

describe("<DateComponent />", function () {
  it("should render with given props when isSelected is true", function () {
    var component = (0, _reactTestRenderer.create)(React.createElement(_DateComponent.DateComponent, {
      date: 1,
      isSelected: true,
      isWeekend: false,
      dateFontSize: 16,
      dateComponentHeight: 50
    })).toJSON();
    expect(component).toMatchSnapshot();
  });
  it("should render with given props when isSelected is false", function () {
    var component = (0, _reactTestRenderer.create)(React.createElement(_DateComponent.DateComponent, {
      date: 1,
      isSelected: false,
      isWeekend: false,
      dateFontSize: 16,
      dateComponentHeight: 50
    })).toJSON();
    expect(component).toMatchSnapshot();
  });
});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi9Vc2Vycy9ha3VybmFkaS9Qcm9qZWN0cy9yZWFjdC1uYXRpdmUtc2NoZWR1bGVyL2xpYnJhcnkvQ2FsZW5kYXJNb250aC9EYXRlQ2FsZW5kYXJCb3gvSW5kaXZpZHVhbENvbXBvbmVudHMvRGF0ZUNvbXBvbmVudC50ZXN0LnRzeCJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOztBQUFBOztBQUNBOztBQUNBOztBQUVBLFFBQVEsQ0FBQyxtQkFBRCxFQUFzQixZQUFBO0FBRTFCLEVBQUEsRUFBRSxDQUFDLHdEQUFELEVBQTJELFlBQUE7QUFDekQsUUFBTSxTQUFTLEdBQUcsK0JBQ2Qsb0JBQUMsNEJBQUQ7QUFDSSxNQUFBLElBQUksRUFBRSxDQURWO0FBRUksTUFBQSxVQUFVLEVBQUUsSUFGaEI7QUFHSSxNQUFBLFNBQVMsRUFBRSxLQUhmO0FBSUksTUFBQSxZQUFZLEVBQUUsRUFKbEI7QUFLSSxNQUFBLG1CQUFtQixFQUFFO0FBTHpCLE1BRGMsRUFPVixNQVBVLEVBQWxCO0FBUUEsSUFBQSxNQUFNLENBQUMsU0FBRCxDQUFOLENBQWtCLGVBQWxCO0FBQ0gsR0FWQyxDQUFGO0FBV0EsRUFBQSxFQUFFLENBQUMseURBQUQsRUFBNEQsWUFBQTtBQUMxRCxRQUFNLFNBQVMsR0FBRywrQkFDZCxvQkFBQyw0QkFBRDtBQUNJLE1BQUEsSUFBSSxFQUFFLENBRFY7QUFFSSxNQUFBLFVBQVUsRUFBRSxLQUZoQjtBQUdJLE1BQUEsU0FBUyxFQUFFLEtBSGY7QUFJSSxNQUFBLFlBQVksRUFBRSxFQUpsQjtBQUtJLE1BQUEsbUJBQW1CLEVBQUU7QUFMekIsTUFEYyxFQU9WLE1BUFUsRUFBbEI7QUFRQSxJQUFBLE1BQU0sQ0FBQyxTQUFELENBQU4sQ0FBa0IsZUFBbEI7QUFDSCxHQVZDLENBQUY7QUFXSCxDQXhCTyxDQUFSIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICogYXMgUmVhY3QgZnJvbSBcInJlYWN0XCJcbmltcG9ydCB7IGNyZWF0ZSB9IGZyb20gXCJyZWFjdC10ZXN0LXJlbmRlcmVyXCJcbmltcG9ydCB7IERhdGVDb21wb25lbnQgfSBmcm9tIFwiLi9EYXRlQ29tcG9uZW50XCJcblxuZGVzY3JpYmUoXCI8RGF0ZUNvbXBvbmVudCAvPlwiLCAoKSA9PiB7XG5cbiAgICBpdChcInNob3VsZCByZW5kZXIgd2l0aCBnaXZlbiBwcm9wcyB3aGVuIGlzU2VsZWN0ZWQgaXMgdHJ1ZVwiLCAoKSA9PiB7XG4gICAgICAgIGNvbnN0IGNvbXBvbmVudCA9IGNyZWF0ZShcbiAgICAgICAgICAgIDxEYXRlQ29tcG9uZW50XG4gICAgICAgICAgICAgICAgZGF0ZT17MX1cbiAgICAgICAgICAgICAgICBpc1NlbGVjdGVkPXt0cnVlfVxuICAgICAgICAgICAgICAgIGlzV2Vla2VuZD17ZmFsc2V9XG4gICAgICAgICAgICAgICAgZGF0ZUZvbnRTaXplPXsxNn1cbiAgICAgICAgICAgICAgICBkYXRlQ29tcG9uZW50SGVpZ2h0PXs1MH1cbiAgICAgICAgICAgIC8+KS50b0pTT04oKVxuICAgICAgICBleHBlY3QoY29tcG9uZW50KS50b01hdGNoU25hcHNob3QoKVxuICAgIH0pXG4gICAgaXQoXCJzaG91bGQgcmVuZGVyIHdpdGggZ2l2ZW4gcHJvcHMgd2hlbiBpc1NlbGVjdGVkIGlzIGZhbHNlXCIsICgpID0+IHtcbiAgICAgICAgY29uc3QgY29tcG9uZW50ID0gY3JlYXRlKFxuICAgICAgICAgICAgPERhdGVDb21wb25lbnRcbiAgICAgICAgICAgICAgICBkYXRlPXsxfVxuICAgICAgICAgICAgICAgIGlzU2VsZWN0ZWQ9e2ZhbHNlfVxuICAgICAgICAgICAgICAgIGlzV2Vla2VuZD17ZmFsc2V9XG4gICAgICAgICAgICAgICAgZGF0ZUZvbnRTaXplPXsxNn1cbiAgICAgICAgICAgICAgICBkYXRlQ29tcG9uZW50SGVpZ2h0PXs1MH1cbiAgICAgICAgICAgIC8+KS50b0pTT04oKVxuICAgICAgICBleHBlY3QoY29tcG9uZW50KS50b01hdGNoU25hcHNob3QoKVxuICAgIH0pXG59KVxuIl19