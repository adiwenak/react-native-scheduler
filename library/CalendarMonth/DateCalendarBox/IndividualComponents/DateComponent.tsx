import * as React from "react"
import { Text, View } from "react-native"
import { DateNumber } from "../../../shared/model"
import styles from "./DateComponent.style"

interface DateComponentProps {
  date?: DateNumber
  isSelected: boolean
  isWeekend: boolean
  dateFontSize: number
  dateComponentHeight: number
}

const getDateTitle = (props: DateComponentProps) => {
  const dateTitle = props.date ? `${props.date}` : ""
  let dateStyle = props.isWeekend ? styles.textStyleWeekend : styles.textStyleWeekday
  if (props.isSelected) {
    dateStyle = styles.titleSelected
  }
  return (
    <Text style={[styles.title, dateStyle, { fontSize: props.dateFontSize }]}>
      {dateTitle}
    </Text>
  )
}

const getSelectedDateTitle = (props: DateComponentProps) => {
  const circleSize = props.dateComponentHeight / 1.2
  const borderRadiusSize = circleSize / 2

  return (
    <View style={[
      styles.selectedColor,
      { height: circleSize, width: circleSize, borderRadius: borderRadiusSize }
    ]}>
      <View style={styles.circleView}>
        {getDateTitle(props)}
      </View>
    </View>
  )
}

export const DateComponent = (props: DateComponentProps) => {
  return (
    <View style={styles.topOuterView}>
      {props.isSelected ? getSelectedDateTitle(props) : getDateTitle(props)}
    </View>
  )
}
