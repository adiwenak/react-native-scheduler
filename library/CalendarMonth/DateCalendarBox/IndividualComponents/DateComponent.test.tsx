import * as React from "react"
import { create } from "react-test-renderer"
import { DateComponent } from "./DateComponent"

describe("<DateComponent />", () => {

    it("should render with given props when isSelected is true", () => {
        const component = create(
            <DateComponent
                date={1}
                isSelected={true}
                isWeekend={false}
                dateFontSize={16}
                dateComponentHeight={50}
            />).toJSON()
        expect(component).toMatchSnapshot()
    })
    it("should render with given props when isSelected is false", () => {
        const component = create(
            <DateComponent
                date={1}
                isSelected={false}
                isWeekend={false}
                dateFontSize={16}
                dateComponentHeight={50}
            />).toJSON()
        expect(component).toMatchSnapshot()
    })
})
