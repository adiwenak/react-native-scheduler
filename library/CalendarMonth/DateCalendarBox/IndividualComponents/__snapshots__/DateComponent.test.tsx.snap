// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateComponent /> should render with given props when isSelected is false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#323A48",
          "fontSize": 14,
          "fontWeight": "500",
        },
        Object {
          "fontSize": 16,
        },
      ]
    }
  >
    1
  </Text>
</View>
`;

exports[`<DateComponent /> should render with given props when isSelected is true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FF1493",
        },
        Object {
          "borderRadius": 20.833333333333336,
          "height": 41.66666666666667,
          "width": 41.66666666666667,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#323A48",
              "fontSize": 14,
              "fontWeight": "500",
            },
            Object {
              "fontSize": 16,
            },
          ]
        }
      >
        1
      </Text>
    </View>
  </View>
</View>
`;
